jujugui/static/gui/src/app/components/deployment-flow/changes/changes.js

```diff
@@ -21,38 +21,49 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
 YUI.add('deployment-changes', function() {
 
   juju.components.DeploymentChanges = React.createClass({
+    displayName: 'DeploymentChanges',
+
     propTypes: {
-      changes: React.PropTypes.object.isRequired,
-      generateAllChangeDescriptions: React.PropTypes.func.isRequired
+      generateAllChangeDescriptions: React.PropTypes.func.isRequired,
+      getCurrentChangeSet: React.PropTypes.func.isRequired
+    },
+
+    getInitialState: function() {
+      this.interval = null;
+      return {
+        changes: this.props.getCurrentChangeSet()
+      };
     },
 
-    /**
-      Generate the list of change items.
+    componentDidMount: function() {
+      // poll the changeset every second for updates as there is no way that
+      // we can be sure when a bundle import will be complete.
+      this.interval = setInterval(() => {
```

Comment:
Ok I'll remove this and then add another indicator as the number of machines this will add in the future to give the user an indication that there might be charges.